### ISSUE TITLE
Deliver current server url as part of view data

### ISSUE THREAD

**Issue by janwidmer** [namics/generator-nitro#154]
## Type of issue

* Feature request

The issue refers to the:

* nitro app (@nitro/app)

As nitro runs on different environments, it would be nice to be able to output the correct server url in the markup, e.g. in the head section.

**Comment by ernscht** [comment]
This can also be easily achieved with a simple [viewData script](https://github.com/namics/generator-nitro/blob/master/packages/project-nitro/project/viewData/readme.md).

```
function getServerUrl(req, res, next) {
    res.locals.pageUrl = `${req.protocol}://${req.get('host')}${req.originalUrl}`;
    next();
}
exports = module.exports = function(app) {
   app.route('*').get(getServerUrl);
};
```

**Comment by janwidmer** [comment]
I am also happy to just solve it via viewData in the project..but I guess, almost every project needs to render the server url in the markup at some point..? What would you prefer?

**Comment by ernscht** [comment]
This was only a tip to solve this in existing projects quickly.
I think the adaptation makes sense.

**Comment by janwidmer** [comment]
ok perfect, I will add a cypress test then..

**Comment by ernscht** [comment]
Released with 4.12.0
